File: server/app/schemas/share.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel

from app.models.enums import ScheduleSource


class ShareScheduleDTO(BaseModel):
    id: int
    title: str
    start_time: datetime
    end_time: datetime | None
    location: str | None
    remark: str | None
    source: ScheduleSource
    updated_at: datetime
    is_deleted: bool


class ShareCreateResponse(BaseModel):
    share_uuid: str
    share_path: str
    schedule: ShareScheduleDTO

File: frontend/src/api/share.ts
import { api } from "@/api/client";

export type ShareScheduleDto = {
  title: string;
  start_time: string;
  end_time: string | null;
  location: string | null;
  remark: string | null;
  source: "manual" | "ai_parsed";
  updated_at: string;
  is_deleted: boolean;
};

export type ShareCreateResponse = {
  share_uuid: string;
  share_path: string;
  schedule: ShareScheduleDto;
};

export async function createShareLink(scheduleId: number): Promise<ShareCreateResponse> {
  const response = await api.post<ShareCreateResponse>(`/share/schedules/${scheduleId}`);
  return response.data;
}

export async function fetchSharedSchedule(shareUuid: string): Promise<ShareScheduleDto> {
  const response = await api.get<ShareScheduleDto>(`/share/${shareUuid}`);
  return response.data;
}
